#!/bin/bash

set -e

cd /opt/stackstorm/st2/lib/python2.7/site-packages/st2actions/notifier

cat << EOF | patch -p0
--- notifier.py 2018-08-21 12:02:08.324739000 +0000
+++ notifier.py.modified        2018-08-21 12:02:23.874739000 +0000
@@ -85,11 +85,6 @@
                 execution_db.status == LIVEACTION_STATUS_PAUSED):
             wf_svc.handle_action_execution_pause(execution_db)

-        if execution_db.status not in LIVEACTION_COMPLETED_STATES:
-            LOG.debug('Skipping processing of execution %s since it\'s not in a completed state' %
-                      (execution_id), extra=extra)
-            return
-
         liveaction_id = execution_db.liveaction['id']
         liveaction_db = LiveAction.get_by_id(liveaction_id)
         self._apply_post_run_policies(liveaction_db=liveaction_db)
EOF

